eng/scripts/linting_suppression_generator.py

```diff
@@ -0,0 +1,224 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# Use case: Generates Checkstyle (checkstyle-suppressions.xml) and Spotbugs (spotbugs-exclude.xml) suppression files
+# for the Java SDK.
+#
+# To use this tool, run the following command:
+#
+# python linting_suppression_generator.py --project-folder <project folder>
```

Comment:
nit: what's a project folder? I assume it is the subdirectories under `sdk\<service>`? One example might help a lot here.

Reply:
Will add that, it'd be `sdk\core\azure-core` not `sdk\core`